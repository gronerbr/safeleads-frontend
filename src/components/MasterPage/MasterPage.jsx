import React, { Fragment, Component } from 'react';
import PropTypes from 'prop-types';
import Grid from 'material-ui/Grid';
import Drawer from 'material-ui/Drawer';
import Header from '../Header/Header';
import Sidebar from '../Sidebar/Sidebar';
import styles from './MasterPage.scss';

class MasterPage extends Component {
  static propTypes = {
    children: PropTypes.arrayOf(PropTypes.element).isRequired,
  }
  constructor(props) {
    super(props);
    window.onresize = () => {
      this.setState({
        isSizeMenuOpen: (document.body.offsetWidth > 900),
      });
    };
    this.state = {
      isClickMenuOpen: false,
      isSizeMenuOpen: (document.body.offsetWidth > 900),
    };
  }
  handleMenuChange() {
    debugger;
  }
  render() {
    return (
      <Fragment>
        <Header onToggleMenu={() => this.setState({ isClickMenuOpen: true })} />
        <Grid container spacing={0}>
          <Grid
            className={styles.drawer}
            item
            xs={10}
            md={2}
          >
            <Drawer
              // open={this.state.isClickMenuOpen || this.state.isSizeMenuOpen}
              open
              variant="persistent"
              classes={{
                docked: {
                  flex: '0 0 auto',
                },
                paper: {
                  top: 100,
                },
              }}
              onClose={() => this.setState({ isClickMenuOpen: false })}
            >
              <Sidebar />
            </Drawer>
          </Grid>
          <Grid
            item
            className={styles.bgContent}
          >
            <div className={`${(this.state.isClickMenuOpen || this.state.isSizeMenuOpen) && styles.baseXsMenuOpen}`}>
              { this.props.children }
              <div className={styles.copyright}>Copyright SafeLeads</div>
            </div>
          </Grid>
        </Grid>
      </Fragment>
    );
  }
}


export default MasterPage;
