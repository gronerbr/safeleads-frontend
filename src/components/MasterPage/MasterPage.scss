.drawer {
  margin-top: 76px;
}
.drawerContainer {
  margin-top: 76px;
}
.menuClosed {
  display: none;
}
.menuOpen {
  background-color: #35414f;
}
.bgContent {
  background-color: #f7f7f7;
  overflow: hidden;
  min-height: calc(100vh - 75px);
}
.copyright {
  color: #8c8c8c;
  padding: 30px 0 80px;
}
@media screen and (max-width: 700px) {
  .baseXsMenuOpen {
    width: 800px;
  }
}
