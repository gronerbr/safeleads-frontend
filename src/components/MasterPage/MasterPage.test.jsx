import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import MasterPage from './MasterPage';

describe('Example: ', () => {
  test('renders without crashing', () => {
    const component = shallow(
        <MasterPage>
          <h1>Test</h1>
        </MasterPage>
      );

    expect(toJson(component)).toMatchSnapshot();
  });
});
