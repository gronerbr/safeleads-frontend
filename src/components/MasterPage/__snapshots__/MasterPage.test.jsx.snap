// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example:  renders without crashing 1`] = `
<React.Fragment>
  <Header
    onToggleMenu={[Function]}
  />
  <WithStyles(GridWrapper)
    container={true}
    spacing={0}
  >
    <WithStyles(GridWrapper)
      item={true}
      md={2}
      xs={10}
    >
      <Sidebar />
    </WithStyles(GridWrapper)>
    <WithStyles(GridWrapper)
      item={true}
      md={10}
      xs={2}
    >
      <div
        className="undefined"
      >
        <h1>
          Test
        </h1>
        <div>
          Copyright SafeLeads
        </div>
      </div>
    </WithStyles(GridWrapper)>
  </WithStyles(GridWrapper)>
</React.Fragment>
`;
